build-system/tasks/extension-generator/index.js

```diff
@@ -274,10 +274,13 @@ module.exports = {
   insertExtensionBundlesConfig,
```

Comment:
Can we add the file when using the legacy generator as well?

Reply:
Done. Also refactored so each generation function just returns the bundle config, and the calling task handles updating that config.